File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/db/IDeckRepository.java
package com.shuneault.netrunnerdeckbuilder.db;

import com.shuneault.netrunnerdeckbuilder.game.Deck;
import com.shuneault.netrunnerdeckbuilder.game.Format;

import java.util.ArrayList;

public interface IDeckRepository {
    Deck getDeck(long deckId);

    ArrayList<Deck> getAllDecks();

    void addDeck(Deck deck);

    boolean deleteDeck(Deck deck);

    void changeIdentity(Deck mDeck, String identityCode);

    Long cloneDeck(Deck deck);

    void createDeck(Deck deck);

    void setDeckFormat(Deck mDeck, Format format);

    void saveDeck(Deck deck);
}

File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/db/IDeckRepository.kt
package com.shuneault.netrunnerdeckbuilder.db

import androidx.lifecycle.MutableLiveData
import com.shuneault.netrunnerdeckbuilder.game.Deck
import com.shuneault.netrunnerdeckbuilder.game.Format
import java.util.ArrayList

interface IDeckRepository {
    fun getDeck(deckId: Long): Deck?
    fun allDecksLiveData(): MutableLiveData<ArrayList<Deck>>
    val allDecks: ArrayList<Deck>
    fun addDeck(deck: Deck)
    fun deleteDeck(deck: Deck): Boolean
    fun changeIdentity(mDeck: Deck, identityCode: String)
    fun cloneDeck(deck: Deck): Long
    fun createDeck(deck: Deck)
    fun setDeckFormat(mDeck: Deck, format: Format)
    fun saveDeck(deck: Deck)
    fun updateDeck(deck: Deck)
}